### PR TITLE
Handles authorisation in image urls with redirection

### PR DESCRIPTION
**Pull request by antonis**
Depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3018

## Description
This PR fixes an issue were images with http redirects do not load due to missing authorisation headers

Internal ref p1716409760977969-slack-C04TBSWP09M

⚠️ Please merge after https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3018 is merged and the FluxC version on this PR is updated to point to trunk.

-----

## To Test:
1. Open the reader and head to a private blog (e.g. a P2)
2. Open a post with images in the comments
3. **Verify** that images with redirects load as expected

-----

## Regression Notes

1. Potential unintended areas of impact

    - Image loading

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

5. What automated tests I added (or what prevented me from doing so)

    - Added test on the FluxC side

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/285">25.0</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ import javax.inject.Named
 @GlideModule
 class WordPressGlideModule : AppGlideModule() {
     @Inject
-    @Named("custom-ssl")
+    @Named("custom-ssl-custom-redirects")
     lateinit var requestQueue: RequestQueue
```
Comment (jarvislin) [inline WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt:28]: I found the `requestQueue` is a param of `VolleyUrlLoader.Factory(requestQueue, glideRequestFactory)`.
And I found https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/AppInitializer.kt#L676 which is also related to Volley. Should we replace the `requestQueue` here with the `@Named("custom-ssl-custom-redirects")` one?

**Review by jarvislin — APPROVED** [review]
The change looks great! I leave a question on the comment, but it should not affect user experience because the function will be triggered after logout.

Please feel free to merge it.

File: WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt (hunk 1 above)
Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt:28]: Good question @jarvislin 👍 
My understanding is the `AppInitializer` `requestQueue` queue handles the api calls and we don't need to implement the custom redirection there. For now we only need the custom redirection for images which are handled by the `GlideModule`.

Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/modules/WordPressGlideModule.kt:28]: Got it! Thanks for the explanation 👍

**Review by thomashorta — APPROVED** [review]
Thanks for the fix @antonis, just took a look at both PRs (FluxC and this) and they look good! 🙇🏼

**Comment by antonis** [comment]
Thank you both for the review 🙇

**Pull request merged**